### PR TITLE
Fix project - integrate fixed MSID

### PR DESCRIPTION
**Pull request by unpluggedk**
Project file was broken, now it is fixed.
https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/155

update after error code PR https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/153

### CODE REVIEW

File: MSAL/src/MSALErrorConverter.m [hunk 1]
```diff
@@ -64,6 +64,7 @@ + (void)initialize
                                    @(MSIDErrorInvalidRequest) :@(MSALErrorInvalidRequest),
                                    @(MSIDErrorInvalidClient) : @(MSALErrorInvalidClient),
                                    @(MSIDErrorInvalidGrant) : @(MSALErrorInvalidParameter),
+                                   @(MSIDErrorInvalidScope) : @(MSALErrorInvalidParameter),
```
Comment (oldalton) [inline MSAL/src/MSALErrorConverter.m]: can we have an MSAL error to return invalid scope? same for invalid grant :)

File: MSAL/src/MSALErrorConverter.m [hunk 2]
```diff
@@ -44,29 +44,35 @@ + (void)initialize
...
                            MSIDOAuthErrorDomain:@{
-                                   @(MSIDErrorInvalidRequest) :@(MSALErrorInvalidRequest),
-                                   @(MSIDErrorInvalidClient) : @(MSALErrorInvalidClient),
-                                   @(MSIDErrorInvalidGrant) : @(MSALErrorInvalidParameter),
-                                   @(MSIDErrorInvalidParameter) : @(MSALErrorInvalidParameter),
-                                   @(MSIDErrorServerRefreshTokenRejected) : @(MSALErrorAuthorizationFailed),
+                                   @(MSIDErrorInteractionRequired) : @(MSALErrorInteractionRequired),
                                    @(MSIDErrorServerOauth) : @(MSALErrorAuthorizationFailed),
+                                   @(MSIDErrorServerInvalidResponse) : @(MSALErrorInvalidResponse),
+                                   @(MSIDErrorServerRefreshTokenRejected) : @(MSALErrorAuthorizationFailed),
```
Comment (oldalton) [inline MSAL/src/MSALErrorConverter.m]: Is there a better corresponding error code than MSALErrorAuthorizationFailed (which sounds more like interactive error)?

File: MSAL/src/MSALErrorConverter.m [hunk 3]
```diff
@@ -44,29 +44,35 @@ + (void)initialize
...
+                                   @(MSIDErrorServerInvalidRequest) :@(MSALErrorInvalidRequest),
+                                   @(MSIDErrorServerInvalidClient) : @(MSALErrorInvalidClient),
+                                   @(MSIDErrorServerInvalidGrant) : @(MSALErrorInvalidGrant),
+                                   @(MSIDErrorServerInvalidScope) : @(MSALErrorInvalidScope),
+                                   @(MSIDErrorServerInvalidState) : @(MSALErrorInvalidState),
+                                   @(MSIDErrorServerNonHttpsRedirect) : @(MSALErrorInternal)
```
Comment (oldalton) [inline MSAL/src/MSALErrorConverter.m]: Don't we have a corresponding MSAL error? If not, let's add it, as it sounds important to call out this specific reason.

File: MSAL/src/MSALErrorConverter.m (hunk 2 above)
Comment (unpluggedk) [reply MSAL/src/MSALErrorConverter.m]: added corresponding error for msal. also filed an issue to improve refactoring #285

File: MSAL/src/MSALErrorConverter.m (hunk 3 above)
Comment (unpluggedk) [reply MSAL/src/MSALErrorConverter.m]: added corresponding error for msal. also filed an issue to improve refactoring #285

File: MSAL/src/MSALErrorConverter.m [hunk 4]
```diff
@@ -44,29 +44,35 @@ + (void)initialize
...
-                                   @(MSIDErrorDeveloperAuthorityValidation) : @(MSALErrorFailedAuthorityValidation),
-                                   @(MSIDErrorServerRefreshTokenRejected) : @(MSALErrorAuthorizationFailed),
-                                   @(MSIDErrorServerOauth) : @(MSALErrorAuthorizationFailed),
-                                   @(MSIDErrorUnsupportedFunctionality): @(MSALErrorInternal)
+                                   // Authority Validation
+                                   @(MSIDErrorAuthorityValidation) : @(MSALErrorFailedAuthorityValidation),
+                                   // Interactive flow
+                                   @(MSIDErrorAuthorizationFailed) : @(MSALErrorAuthorizationFailed),
+                                   @(MSIDErrorUserCancel) : @(MSALErrorUserCanceled),
+                                   @(MSIDErrorSessionCanceledProgramatically) : @(MSALErrorSessionCanceled),
```
Comment (oldalton) [inline MSAL/src/MSALErrorConverter.m]: nit: MSIDErrorSessionCanceledProgramatically should have double "m"

**Pull request merged**